src/index.ts

```diff
@@ -7,15 +7,38 @@ import CollectorApi from './collector/api';
 
 export async function printOnce(): Promise<void> {
 	const opts = getOptions();
+
 	const collector = CollectorApi.getCollector();
+	logger.info('Collector obtained successfully');
 
 	if (opts.autoDiscover) {
+		logger.info('Starting auto-discovery...');
 		await collector.discoverAll();
+		logger.info('Auto-discovery completed');
 	}
 
+	logger.info('Updating metrics...');
 	await collector.updateAll();
-	await collector.close();
+	logger.info('Metrics updated successfully');
+
+	logger.info('Closing collector...');
+	const closePromise = collector.close();
+	const timeoutPromise = new Promise((_, reject) =>
+		setTimeout(() => reject(new Error('Close timeout')), 2000)
+	);
 
+	try {
+		await Promise.race([closePromise, timeoutPromise]);
+		logger.info('Collector closed successfully');
+	} catch (error) {
+		logger.warn('Collector close timed out or failed:', error);
+		// Force disconnect the Redis client
+		logger.info('Force disconnecting Redis client...');
+		collector['defaultRedisClient'].disconnect();
```

Comment:
prob no need to do this, if we're relying on `process.exit` to exit an app. if a timeout occurs and is "handled" here - the following `process.exit(0)` will essentially kill -9 everything in node that hasn't been awaited before it

that said, `process.exit(0)` is discouraged for graceful exit precisely because it nukes the app